### PR TITLE
HTTPSify links

### PR DESCRIPTION
**Pull request by MikkCZ**
Where possible, use https or hardcoded links (these http URLs mostly result in 301 redirect anyway).

### CODE REVIEW

File: docs/hacking_howto.rst [hunk 1]
```diff
@@ -82,7 +82,7 @@ These are required for the minimum installation:
 
 * git
 * Python 2.7
-* pip: `<http://www.pip-installer.org/en/latest/>`_
+* pip: `<https://www.pip-installer.org/en/latest/>`_
```
Comment (mythmon) [inline docs/hacking_howto.rst]: This link doesn't work. I imagine pip-installer.org isn't using HTTPS.

File: docs/installation-vagrant.rst [hunk 2]
```diff
@@ -31,7 +31,7 @@ Install and run everything
       vagrant up
 
    .. note:: VirtualBox creates VMs in your system drive.
-             If it won't fit on your system drive, you will need to `change that directory to another drive <http://emptysquare.net/blog/moving-virtualbox-and-vagrant-to-an-external-drive/>`_.
+             If it won't fit on your system drive, you will need to `change that directory to another drive <https://emptysquare.net/blog/moving-virtualbox-and-vagrant-to-an-external-drive/>`_.
```
Comment (mythmon) [inline docs/installation-vagrant.rst]: This link also doesn't work.

**Comment by MikkCZ** [comment]
Deep apology for the broken links, @mythmon . I let messed my mind by redirects. Corrected now.

**Comment by mythmon** [comment]
These changes look good now, but there are merge conflicts. Can you rebase this on master to fix those?

**Comment by MikkCZ** [comment]
Rebased. :)

**Comment by mythmon** [comment]
@homu r+

**Comment by mythmon** [comment]
Master is broken right now. I'll fix that.

**Comment by safwanrahman** [comment]
@homu test

**Comment by safwanrahman** [comment]
@homu retry

**Comment by mythmon** [comment]
@homu retry

**Comment by mythmon** [comment]
@homu retry

**Pull request merged**